src/lib/y2partitioner/clients/main.rb

```diff
@@ -23,25 +25,33 @@ module Clients
     class Main
       extend Yast::I18n
       extend Yast::UIShortcuts
+      extend Yast::Logger
 
       # Run the client
       def self.run
         textdomain "storage"
 
-        probed = Y2Storage::StorageManager.instance.y2storage_probed
-        staging = Y2Storage::StorageManager.instance.y2storage_staging
-        DeviceGraphs.instance.original = probed
-        DeviceGraphs.instance.current = staging
-        overview_w = CWM::TreePager.new(Widgets::OverviewTree.new(staging))
-
-        contents = MarginBox(
-          0.5,
-          0.5,
-          overview_w
-        )
+        DeviceGraphs.instance.original = Y2Storage::StorageManager.instance.y2storage_probed
+        DeviceGraphs.instance.current = Y2Storage::StorageManager.instance.y2storage_staging.dup
 
         Yast::Wizard.CreateDialog unless Yast::Stage.initial
-        Yast::CWM.show(contents, caption: _("Partitioner"))
+        res = nil
+        loop do
+          contents = MarginBox(
+            0.5,
+            0.5,
+            CWM::TreePager.new(Widgets::OverviewTree.new(DeviceGraphs.instance.current))
+          )
+          res = Yast::CWM.show(contents, caption: _("Partitioner"), skip_store_for: [:redraw])
+          break if res != :redraw
+        end
+
+        # Running system: presenting "Expert Partitioner: Summary" step now
+        # ep-main.rb SummaryDialog
+        if res == :next && Yast::Popup.ContinueCancel("(potentially) d3stR0y Ur DATA?!??")
+          Y2Storage::StorageManager.instance.staging = DeviceGraphs.instance.current
+          Y2Storage::StorageManager.instance.commit
```

Comment:
Good point. I haven't tested this in the installation at all, and I think we have ignored the issue during scrum planning.